stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/MessageFooterDecorator.kt

```diff
@@ -92,15 +95,18 @@ internal class MessageFooterDecorator(
     }
 
     private fun setupMessageFooterTime(textView: TextView, data: MessageListItem.MessageItem) {
+        fun TextView.showTime(time: Date) {
+            isVisible = true
+            text = dateFormatter.formatTime(time)
+        }
+
         val createdAt = data.message.getCreatedAtOrNull()
+        val updatedAt = data.message.getUpdatedAtOrNull()
+
         when {
             data.isNotBottomPosition() || createdAt == null -> textView.isVisible = false
-            else -> {
-                textView.apply {
-                    isVisible = true
-                    text = dateFormatter.formatTime(createdAt)
-                }
-            }
+            data.message.isGiphyNotEphemeral() && updatedAt != null -> textView.showTime(updatedAt)
```

Comment:
If it's not ephemeral, but `updatedAt == null`, I think we're just not doing anything with the view, so it'll remain in the previously bound state. I think we should handle that case, at least hide the view or something.

Reply:
For your described case old logic shows `createdAt` time. I think it's better than nothing. Also not clear why for regular giphy message updatedAt is null

Reply:
You're right, that works 👍